varinfo/cf_config.py

```diff
@@ -160,40 +155,43 @@ def _create_attributes_object(cf_item: dict) -> dict[str, str]:
             for attribute in cf_item.get('Attributes', {})
         }
 
-    def get_cf_attributes(
-        self, variable: str | None = None
-    ) -> dict[str, dict[str, str]]:
-        """Return the CF overrides and supplements that match a given
-        variable. If a variable is not specified, then return all overrides
-        and supplements. If there are no overrides or supplements, then
-        empty dictionaries will be returned instead.
+    def get_cf_overrides(self, variable_path: str) -> dict[str, Any]:
+        """Return the CF overrides that match a given variable. If there are no
+        overrides, then empty dictionaries will be returned instead.
 
-        """
-        if variable is not None:
-            cf_overrides = self._get_matching_attributes(self._cf_overrides, variable)
-            cf_supplements = self._get_matching_attributes(
-                self._cf_supplements, variable
-            )
-        else:
-            cf_overrides = self._cf_overrides
-            cf_supplements = self._cf_supplements
+        First iterate through the self.cf_overrides and find all items with a
+        variable pattern that matches the supplied variable (or group) path.
 
-        return {'cf_overrides': cf_overrides, 'cf_supplements': cf_supplements}
+        Next sort that dictionary, so that variable pattern keys are ordered
+        from the shortest (more generic) to longest (more specific).
 
-    @staticmethod
-    def _get_matching_attributes(
-        cf_references: dict[str, dict[str, str]], variable: str
-    ) -> dict[str, str]:
-        """Iterate through either the self._cf_supplements or
-        self._cf_overrides and extract a dictionary that combines all
-        applicable attributes that apply to the specified variable. If
-        there are conflicting values for the same attribute, only the last
-        value will be returned for that attribute.
+        Last, combine the attribute names and values from each matching
+        override item. Because of the ordering in the previous step, if there
+        are multiple values supplied for the same metadata attribute, the value
+        retained will be the one with the longest variable pattern, which is a
+        proxy for how specific the override is.
 
         """
```

Comment:
Now that there are only overrides, and not supplements, there isn't a need to break out this functionality. Also the reduction in complexity of `CFConfig.get_cf_overrides` means that combining the two methods is still pretty clean.